### PR TITLE
OSSM-8338: Update gateway content

### PR DESCRIPTION
**Pull request by rh-tokeefe**
Affects:

[service-mesh-docs-main](https://github.com/openshift/openshift-docs/tree/service-mesh-docs-main)
[service-mesh-docs-3.0.0tp1](https://github.com/openshift/openshift-docs/tree/service-mesh-docs-3.0.0tp1)

This PR is part of the standalone doc set for the Service Mesh 3.0 project. Kathryn is aware that this content applies for a product that is part of a Technology Preview release. The project is seeking feedback from early adopters.

PR must be CP'd back to the tp1 branch.

Version(s): Tech Preview

Issue: https://issues.redhat.com/browse/OSSM-8338

Link to docs preview:
https://84361--ocpdocs-pr.netlify.app/openshift-service-mesh/latest/gateways/ossm-getting-traffic-into-a-mesh.html

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc
Comment (eoinfennessy) [inline modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc:193]: The same formatting issue mentioned earlier is also present here.

File: gateways/ossm-getting-traffic-into-a-mesh.adoc
Comment (eoinfennessy) [inline gateways/ossm-getting-traffic-into-a-mesh.adoc:24]: Should this be an xref? Or a link to the latest OCP docs instead of fixed to version 4.17?

**Review by eoinfennessy — COMMENTED** [review]
Looks good. I added a few minor comments.

File: modules/ossm-about-configuring-a-gateway-to-accept-ingress-traffic.adoc
Comment (rh-tokeefe) [reply modules/ossm-about-configuring-a-gateway-to-accept-ingress-traffic.adoc:38]: done

File: modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc
Comment (rh-tokeefe) [reply modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc:36]: done

File: modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc
Comment (rh-tokeefe) [reply modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc:117]: done

File: modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc
Comment (rh-tokeefe) [reply modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc:173]: done

File: modules/ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes.adoc
Comment (rh-tokeefe) [reply modules/ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes.adoc]: The formatting looks like it was addressed

File: modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc
Comment (eoinfennessy) [inline modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc:31]:
```suggestion
. Deploy a sample service named `httpbin` by applying the `httpbin.yaml` file: 
```

**Review by eoinfennessy — COMMENTED** [review]
Looks good. I just suggested one simple change.

In this PR, can you please also reintroduce the missing triple hyphens in step 2 of the "Installing a gateway by using gateway injection" procedure?

File: modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc
Comment (rh-tokeefe) [reply modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc:31]: done

File: modules/ossm-installing-gateway-using-gateway-injection.adoc
Comment (eoinfennessy) [inline modules/ossm-installing-gateway-using-gateway-injection.adoc:55]: This one should be moved up a line.

**Review by eoinfennessy — APPROVED** [review]
Looks good!

File: modules/ossm-installing-gateway-using-gateway-injection.adoc
Comment (rh-tokeefe) [reply modules/ossm-installing-gateway-using-gateway-injection.adoc:54]: done

**Review by pbajjuri20 — APPROVED** [review]
LGTM..!

File: modules/ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes.adoc
Comment (skopacz1) [inline modules/ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes.adoc:46]: Totally up to you but it might follow our guidelines more closely (and tell users the important action sooner) if you say to run a command at the end:
```suggestion
.. If the cluster runs on AWS,  set the `INGRESS_HOST` variable by running the following command:
```

If you do decide to go with that suggestion, please also apply it to the next substep

File: modules/ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes.adoc
Comment (skopacz1) [inline modules/ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes.adoc:67]: This is arguably another instruction for users to follow and should be its own step:
```suggestion
. Verify that the response has the `HTTP/1.1 200 OK` status, which indicates that the request was successful.
```

File: modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc
Comment (skopacz1) [inline modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc:28]: Out of scope and totally up to you if you're game for content improvements, but these first two steps could use the addition of "by running the following command:"

File: modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc
Comment (skopacz1) [inline modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc:31]: This also needs that command language, but already has "by [doing some action]" wording. I'll leave it up to you to figure out, but maybe this could work:
```suggestion
. Apply the `httpbin.yaml` file to deploy a sample service named `httpbin` by running the following command: 
```
Not sure I'm satisfied with that, but you get the point and I trust that you know what's right. Maybe one of the "apply the yaml file" or "deploy a service" phrases can be dropped and still have the step be reasonably informative. 

Or maybe you can say "this deploys a service named X" at the end, even though we generally should avoid describing the results of steps like that

File: modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc
Comment (skopacz1) [inline modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc:109]: The wording of this step could use the same language if you want

File: modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc
Comment (skopacz1) [inline modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc:112]: Same situation as earlier with this command. IMO, now that I'm reading a second command just like this, I feel like you can drop the "by applying the X yaml file" phrase from both of these steps. 

It seems like the name of the YAML file is not relevant given that it's only part of the filepaths in these commands, and users otherwise might not need to know about the file names more than the action they're taking

File: modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc
Comment (skopacz1) [inline modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc:119]:
```suggestion
. Set a `CURL_POD` variable with the name of the curl pod by running the following command:
```

File: modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc
Comment (skopacz1) [inline modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc:126]: This line should also mention the command language somewhere in the first sentence, though how you do that is up to you. Idk enough about the context of this to suggest what to move around or leave out

File: modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc
Comment (skopacz1) [inline modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc:129]: These lines need to be switched so that the source block isn't in the actual command syntax:
```suggestion
[source,terminal]
----
```

File: modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc
Comment (skopacz1) [reply modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc:126]: Also, at this point I'll stop making this comment and just ask that you add this language at least to any applicable step you touch in your PR.

File: modules/ossm-installing-gateway-using-gateway-injection.adoc
Comment (skopacz1) [inline modules/ossm-installing-gateway-using-gateway-injection.adoc:1]: This is not something that needs to be addressed in this PR, but just wanted to share that it's a little odd to have this procedure included in an "About X" page. Typically, about pages are reserved for conceptual content that gives an overview of a topic, and procedures should be put in procedural assemblies that are titled stuff like "Performing X action".

**Review by skopacz1 — COMMENTED** [review]
Left some comments, otherwise LGTM! Let me know if you have any questions

File: modules/ossm-installing-gateway-using-gateway-injection.adoc
Comment (eoinfennessy) [reply modules/ossm-installing-gateway-using-gateway-injection.adoc:44]: This is valid syntax that allows multiple YAML resources to be contained in one file.

I have seen that this renders with an underline in other places in OpenShift Docs. I don't know why this is the case -- maybe it is intended to make a clearer separation between YAML resources? The underline is not copied when copying the code block, so it should have no effect on usage.

File: modules/ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes.adoc
Comment (rh-tokeefe) [reply modules/ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes.adoc:13]: done

File: modules/ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes.adoc
Comment (rh-tokeefe) [reply modules/ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes.adoc:17]: done

File: modules/ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes.adoc
Comment (rh-tokeefe) [reply modules/ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes.adoc:46]: done

File: modules/ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes.adoc
Comment (rh-tokeefe) [reply modules/ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes.adoc:67]: done

File: modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc
Comment (rh-tokeefe) [reply modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc:28]: done

File: modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc
Comment (rh-tokeefe) [reply modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc]: done

File: modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc
Comment (rh-tokeefe) [reply modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc]: done

File: modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc
Comment (rh-tokeefe) [reply modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc]: done

File: modules/ossm-installing-gateway-using-gateway-injection.adoc
Comment (rh-tokeefe) [reply modules/ossm-installing-gateway-using-gateway-injection.adoc:1]: The intent is to have the procedure after the conceptual topic, with both residing in the same assembly.  I'll discuss with the developer and see if there is anything we want to change here.

File: gateways/ossm-getting-traffic-into-a-mesh.adoc
Comment (eoinfennessy) [reply gateways/ossm-getting-traffic-into-a-mesh.adoc]: The additional resource linking to the OpenShift routes page is relevant to the procedure _Exposing a gateway to traffic outside the cluster by using OpenShift Routes_.

The additional resource linking to the _About gateway injection_ section is relevant to the _About configuring a gateway installed using gateway injection to accept ingress traffic_ conceptual content.

Ideally each could  as part of the section they are associated with, below the main content. Although if this is not possible what is currently rendered seems fine too.

File: modules/ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes.adoc
Comment (eoinfennessy) [inline modules/ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes.adoc:24]:
```suggestion
. Create a YAML file named `httpbin-route.yaml` that defines a `Route` for the `httpbin` service.
```

**Review by eoinfennessy — APPROVED** [review]
I've suggested one minor change. Otherwise, this looks good. Thanks Tim!

File: modules/ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes.adoc
Comment (rh-tokeefe) [reply modules/ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes.adoc:24]: done

File: gateways/ossm-getting-traffic-into-a-mesh.adoc
Comment (rh-tokeefe) [reply gateways/ossm-getting-traffic-into-a-mesh.adoc]: Thanks; done.

**Pull request merged**
